pyiceberg/catalog/__init__.py

```diff
@@ -74,6 +75,8 @@
 LOCATION = "location"
 EXTERNAL_TABLE = "EXTERNAL_TABLE"
 
+TABLE_METADATA_FILE_NAME_REGEX = re.compile(r"""\d*-.*\.json""", re.X)
```

Comment:
Two minor suggestions:

- Change `*` to `+` to have at least one digit.
- Use a group to match the version so we avoid having to do a split on the `-`

```suggestion
TABLE_METADATA_FILE_NAME_REGEX = re.compile(r"""(\d+)-.*\.json""", re.X)
```

Reply:
Thanks for the suggestion!